File: services/webshop-frontend/src/app/shared/models/vehicle.ts
export type VehicleType =
  | 'CAR'
  | 'VAN'
  | 'SUV'
  | 'MOTORBIKE';

export interface Vehicle {
  id: number;
  type: VehicleType;
  brand: string;
  model: string;
  pricePerDay: number;
  description: string;
}

File: services/webshop/src/main/java/com/sep/webshop/entity/Vehicle.java
package com.sep.webshop.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "vehicles")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VehicleType type;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private double pricePerDay;

    @Column(nullable = false)
    private boolean active = true;

    private String description;

}
